File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.java
// Last updated: 3/30/2026, 2:13:16 PM
class Solution {
    public int firstUniqChar(String s) {
        LinkedHashMap<Character,Integer> map = new LinkedHashMap<>();
        for(char ch : s.toCharArray()) map.put(ch,map.getOrDefault(ch,0)+1);
        for(Character key : map.keySet()){
            if(map.get(key)==1) return s.indexOf(key); 
        }
        return -1;
    }
}

File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.c
// Last updated: 3/30/2026, 2:13:10 PM
int firstUniqChar(char* s) {
    int c[26] = {0};
    for(int i = 0 ; i < strlen(s) ; i++){
        c[s[i]-'a']++;
    }
    for(int i = 0 ; i < strlen(s) ; i++){
        if(c[s[i]-'a']==1){
            char *e = strchr(s,s[i]);
            return (int)(e - s);
        }
    }
    return -1;
}